contracts/gmp-express/ExpressProxyFactory.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+import { IExpressProxyDeployer } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IExpressProxyDeployer.sol';
+import { AxelarExecutable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol';
+import { AddressToString, StringToAddress } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/AddressString.sol';
+import { IAxelarGasService } from '../interfaces/IAxelarGasService.sol';
+import { IExpressProxyFactory } from '../interfaces/IExpressProxyFactory.sol';
+
+contract ExpressProxyFactory is AxelarExecutable, IExpressProxyFactory {
+    using AddressToString for address;
+    using StringToAddress for string;
+
+    enum Command {
+        DeployExpressExecutable
+    }
+
+    IAxelarGasService public immutable gasService;
+    IExpressProxyDeployer public immutable proxyDeployer;
+
+    constructor(
+        address gateway_,
+        address gasService_,
+        address proxyDeployer_
+    ) AxelarExecutable(gateway_) {
+        if (gasService_ == address(0)) revert InvalidAddress();
+        if (proxyDeployer_ == address(0)) revert InvalidAddress();
+
+        gasService = IAxelarGasService(gasService_);
+        proxyDeployer = IExpressProxyDeployer(proxyDeployer_);
+    }
+
+    function isExpressProxy(address proxyAddress) public view returns (bool) {
+        return proxyDeployer.isExpressProxy(proxyAddress);
+    }
+
+    function deployedProxyAddress(bytes32 salt, address sender) external view returns (address) {
+        return proxyDeployer.deployedProxyAddress(salt, sender, address(this));
+    }
+
+    function deployExpressProxy(
```

Comment:
Anyone can deploy?

Reply:
Also, what's the use case for this function? Why are there 2 different deployment functions, one deploys only the proxy and one deploys both the proxy and executable?

Reply:
Anyone can deploy their proxy through the service and be sure it will be validated for express execution. If you already have your implementation deployed you just need to deploy proxy and point to it. But if you have nothing deployed it's very convenient and cheaper to deploy proxy + implementation in one TX.